src/compas_timber/_fabrication/jack_cut.py

```diff
@@ -0,0 +1,276 @@
+import math
+
+from compas.geometry import BrepTrimmingError
+from compas.geometry import Frame
+from compas.geometry import Line
+from compas.geometry import Plane
+from compas.geometry import Rotation
+from compas.geometry import Vector
+from compas.geometry import angle_vectors_signed
+from compas.geometry import distance_point_point
+from compas.geometry import intersection_line_plane
+from compas.geometry import is_point_behind_plane
+
+from compas_timber.elements import FeatureApplicationError
+
+
+class BTLxProcess(object):
+    """Base class for BTLx processes.
+
+    Attributes
+    ----------
+    ref_side_index : int
+        The reference side, zero-based, index of the beam to be cut. 0-5 correspond to RS1-RS6.
+    """
+
+    def __init__(self, ref_side_index):
+        self.ref_side_index = ref_side_index
+
+
+class OrientationType(object):
+    """Enum for the orientation of the cut.
+
+    Attributes
+    ----------
+    START : int
+        The start of the beam is cut away.
+    END : int
+        The end of the beam is cut away.
+    """
+
+    START = 0
+    END = 1
+
+
+class JackRafterCut(BTLxProcess):
+    """Represents a Jack Rafter Cut feature to be made on a beam.
+
+    Parameters
+    ----------
+    orientation : int
+        The orientation of the cut. Must be either OrientationType.START or OrientationType.END.
+    start_x : float
+        The start x-coordinate of the cut in parametric space of the reference side. -100000.0 < start_x < 100000.0.
+    start_y : float
+        The start y-coordinate of the cut in parametric space of the reference side. 0.0 < start_y < 50000.0.
+    start_depth : float
+        The start depth of the cut. 0.0 < start_depth < 50000.0.
+    angle : float
+        The horizontal angle of the cut. 0.1 < angle < 179.9.
+    inclination : float
+        The vertical angle of the cut. 0.1 < inclination < 179.9.
+
+    """
+
+    def __init__(self, orientation, start_x=0.0, start_y=0.0, start_depth=0.0, angle=90.0, inclination=90.0, **kwargs):
+        super(JackRafterCut, self).__init__(**kwargs)
+        self._orientation = None
+        self._start_x = None
+        self._start_y = None
+        self._start_depth = None
+        self._angle = None
+        self._inclination = None
+
+        self.orientation = orientation
+        self.start_x = start_x
+        self.start_y = start_y
+        self.start_depth = start_depth
+        self.angle = angle
+        self.inclination = inclination
+
+    @property
+    def orientation(self):
+        return self._orientation
+
+    @orientation.setter
```

Comment:
what is the benefit of using setters and private properties for this class?

Reply:
this an object oriented thing that's strongly related to the encapsulation principle. In statically typed languages where private access is enforced you'd typically always need setters and getters. In python, since nothing is enforced by the language, it can get blurry.

There's different reasons why you'd want to use properties in python. In this particular case, I used properties and their setters for input validation. Since BTLx has a clear specification for the allowed value ranges for each parameter it seemed like a good place to validate those.